triton-vm/src/fri.rs

```diff
@@ -166,7 +166,7 @@ impl ProverRound {
     }
 
     fn merkle_tree_from_codeword(codeword: &[XFieldElement]) -> ProverResult<MerkleTree> {
-        let digests = codeword_as_digests(codeword);
+        let digests: Vec<_> = codeword.par_iter().map(|&xfe| xfe.into()).collect();
         MerkleTree::new::<CpuParallel>(&digests).map_err(FriProvingError::MerkleTreeError)
```

Comment:
We will need a single-threaded Merkle tree builder too, no?

Reply:
The line you highlight is in the prover. I have removed the parallelism from the function `codeword_as_digests` and inlined the parallel version that is used in the prover.

Reply:
Oh, I think I misunderstood your comment. Let me check the code again.

Reply:
You are absolutely right. The FRI verifier is building a Merkle tree from the last round's codeword, and the only logic for building Merkle trees we currently have is using parallelism.[^1]

[^1]: Getting into the nitty gritty details, upon which the FRI verifier should absolutely not depend, Merkle trees with fewer than 512 leafs are built in sequence (unless a rather specific environment variable overrides the default).

As a consequence, we seem to require a new Merkle tree builder, `CpuSequential`. Would you agree?

Reply:
> we seem to require a new Merkle tree builder, CpuSequential. Would you agree?

Yes please!

I also noticed one place in some guesser where we use that. Since it's only 8 leafs there, it probably defaults to sequential anyway. It would be good to call the right method directly though, and not have to rely on contextual knowledge.

Reply:
It seems that the code for FRI is not as deduplicated as it could be; `merkle_tree_from_codeword` is a function associated with `ProverRound`, and is never called in the verifier, even though the verifier performs the same operations in its method `last_round_codeword_merkle_root`. Thanks for the heads up, though!